### PR TITLE
[Driver, Easy] Refactoring deposit_driver for readability

### PR DESCRIPTION
**Pull request by josojo**
refactoring with hopefully the same beauty as Felix's code

### CODE REVIEW

File: driver/src/deposit_driver.rs [hunk 1]
```diff
@@ -21,74 +23,41 @@ pub fn run_deposit_listener<D, C>(db: &D, contract: &C) -> Result<(bool), Driver
...
+            let updated_balances = apply_deposits(&mut balances, &deposits);
+            let new_state_root = H256::from(updated_balances.rolling_hash());
+            
+            println!("New State_hash is {}", new_state_root);
+            contract.apply_deposits(slot, state_root, new_state_root, contract_deposit_hash)?;
             return Ok(true);
+        } else {
+            println!("Need to wait before processing deposit_slot {:?}", slot);
         }
-    } else {
```
Comment (fleupold) [inline driver/src/deposit_driver.rs]: I kind of liked this piece of information

File: driver/src/deposit_driver.rs [hunk 2]
```diff
@@ -21,74 +23,41 @@ pub fn run_deposit_listener<D, C>(db: &D, contract: &C) -> Result<(bool), Driver
...
+            let state_root = contract.get_current_state_root()?;
+            let contract_deposit_hash = contract.deposit_hash_for_slot(slot)?;
+            let mut balances = db.get_current_balances(&state_root)?;
+
+            let deposits = db.get_deposits_of_slot(slot.low_u32())?;
+            let deposit_hash = deposits.rolling_hash();
+            if deposit_hash != contract_deposit_hash {
+                return Err(DriverError::new(
+                    &format!("Pending deposit hash from contract ({}), didn't match the one found in db ({})", 
+                    deposit_hash, contract_deposit_hash), ErrorKind::StateError
```
Comment (fleupold) [inline driver/src/deposit_driver.rs]: The order here is off. According to the string we have to print `contract_deposit_hash` first

File: driver/src/deposit_driver.rs [hunk 3]
```diff
@@ -21,74 +23,41 @@ pub fn run_deposit_listener<D, C>(db: &D, contract: &C) -> Result<(bool), Driver
...
+                ));
+            }
+
+            let updated_balances = apply_deposits(&mut balances, &deposits);
```
Comment (fleupold) [inline driver/src/deposit_driver.rs]: I find it weird that this method returns state *and* takes mutable state (hinting that the state that is input is changed and changed state is returned). 

I’d prefer if the input is immutable and we just return the updated state.

File: driver/src/deposit_driver.rs (hunk 2 above)
Comment (bh2smith) [inline driver/src/deposit_driver.rs]:
```suggestion
                    contract_deposit_hash, deposit_hash), ErrorKind::StateError
```

**Pull request merged**